app/handler/post.go

```diff
@@ -0,0 +1,266 @@
+			return
+		}
+		if q, _ := ctx.Query("offset"); q != "" {
+			offset, err = strconv.Atoi(q)
+		}
+		if err != nil {
+			ctx.SendStatus(http.StatusBadRequest)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		if q, _ := ctx.Query("limit"); q != "" {
+			limit, err = strconv.Atoi(q)
+		}
+		if err != nil {
+			ctx.SendStatus(http.StatusBadRequest)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		published, _ := ctx.Query("published")
+		switch published {
+		case "true":
+			posts, err = model.GetPublishedPosts(offset, limit)
+		case "false":
+			posts, err = model.GetUnpublishedPosts(offset, limit)
+		default:
+			posts, err = model.GetAllPosts(offset, limit)
+		}
+		if err != nil {
+			ctx.SendStatus(http.StatusInternalServerError)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		ctx.JSON(NewAPISuccessResponse(posts))
+	}
+}
+
+// APIPostCommentsHandler gets the comments on the given post.
+func APIPostCommentsHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	comments := post.Comments()
+	ctx.JSON(NewAPISuccessResponse(comments))
+}
+
+// APIPostAuthorHandler gets the author of the given post.
+func APIPostAuthorHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	author := post.Author()
+	ctx.JSON(NewAPISuccessResponse(author))
+}
+
+// APIPostExcerptHandler gets the excerpt of the given post.
+func APIPostExcerptHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	excerpt := post.Excerpt()
+	ctx.JSON(NewAPISuccessResponse(excerpt))
+}
+
+// APIPostSummaryHandler gets the summary of the given post.
+func APIPostSummaryHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	summary := post.Summary()
+	ctx.JSON(NewAPISuccessResponse(summary))
+}
+
+// APIPostTagStringHandler gets the tag string of the given post.
+func APIPostTagStringHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	tagString := post.TagString()
+	ctx.JSON(NewAPISuccessResponse(tagString))
+}
+
+// APIPostTagsHandler gets the tags of the given post.
+func APIPostTagsHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	tags := post.Tags()
+	ctx.JSON(NewAPISuccessResponse(tags))
+}
+
+// APIPostSaveHandler saves the post given in the json-formatted request body.
+func APIPostSaveHandler(ctx *golf.Context) {
+	token, err := ctx.Session.Get("jwt")
+	if err != nil {
+		ctx.SendStatus(http.StatusInternalServerError)
+		return
+	}
+	post := model.NewPost()
+	post.CreatedBy = token.(model.JWT).UserID
+	defer ctx.Request.Body.Close()
+	body, err := ioutil.ReadAll(ctx.Request.Body)
+	if err != nil {
+		utils.LogOnError(err, "Unable to update post from request JSON.", true)
```

Comment:
Yeah I think 500 would be good, although I can't think of a single time when I've seen this actually return the error so it shouldn't really matter haha